src/screens/Onboarding/StepProfile/index.tsx

```diff
@@ -1,55 +1,317 @@
-  const {dispatch} = React.useContext(Context)
+  const t = useTheme()
+  const {gtMobile} = useBreakpoints()
+  const {track} = useAnalytics()
+  const {requestPhotoAccessIfNeeded} = usePhotoLibraryPermission()
+  const creatorControl = Dialog.useDialogControl()
+  const [error, setError] = React.useState('')
+
+  const {state, dispatch} = React.useContext(Context)
+  const [avatar, setAvatar] = React.useState<Avatar>({
+    image: state.profileStepResults?.image,
+    placeholder: emojiItems.at,
+    backgroundColor: randomColor,
+    useCreatedAvatar: false,
+  })
+
+  const canvasRef = React.useRef<PlaceholderCanvasRef>(null)
+
+  React.useEffect(() => {
+    track('OnboardingV2:StepProfile:Start')
+  }, [track])
+
+  const openPicker = React.useCallback(
+    async (opts?: ImagePickerOptions) => {
+      const response = await launchImageLibraryAsync({
+        exif: false,
+        mediaTypes: MediaTypeOptions.Images,
+        quality: 1,
+        ...opts,
+      })
+
+      return (response.assets ?? [])
+        .slice(0, 1)
+        .filter(asset => {
+          if (
+            !asset.mimeType?.startsWith('image/') ||
+            (!asset.mimeType?.endsWith('jpeg') &&
+              !asset.mimeType?.endsWith('jpg') &&
+              !asset.mimeType?.endsWith('png'))
+          ) {
+            setError(_(msg`Only .jpg and .png files are supported`))
+            return false
+          }
+          return true
+        })
+        .map(image => ({
+          mime: 'image/jpeg',
+          height: image.height,
+          width: image.width,
+          path: image.uri,
+          size: getDataUriSize(image.uri),
+        }))
+    },
+    [_, setError],
+  )
+
+  const onContinue = React.useCallback(async () => {
+    let imageUri = avatar?.image?.path
+    if (!imageUri || avatar.useCreatedAvatar) {
+      imageUri = await canvasRef.current?.capture()
+    }
+
+    if (imageUri) {
+      dispatch({
+        type: 'setProfileStepResults',
+        image: avatar.image,
+        imageUri,
+        imageMime: avatar.image?.mime ?? 'image/jpeg',
+      })
+    }
 
-  const onContinue = React.useCallback(() => {
     dispatch({type: 'next'})
-  }, [dispatch])
+    track('OnboardingV2:StepProfile:End')
+  }, [avatar.image, avatar.useCreatedAvatar, dispatch, track])
+
+  const onDoneCreating = React.useCallback(() => {
+    setAvatar(prev => ({
+      ...prev,
+      useCreatedAvatar: true,
+    }))
+    creatorControl.close()
+  }, [creatorControl])
+
+  const openLibrary = React.useCallback(async () => {
+    if (!(await requestPhotoAccessIfNeeded())) {
+      return
+    }
+
+    setError('')
+
+    const items = await openPicker({
+      aspect: [1, 1],
+    })
+    let image = items[0]
+    if (!image) return
+
+    if (!isWeb) {
+      image = await openCropper({
+        mediaType: 'photo',
+        cropperCircleOverlay: true,
+        height: image.height,
+        width: image.width,
+        path: image.path,
```

Comment:
This is confusing to me (we use this same logic in `UserAvatar` as well so not specific to this PR, I think I copied it from UserAvatar before, so we should fix both).

A photo that I pick can be any size, but our avatar should always have a 1:1 aspect ratio - and probably should have some capped size like 250x250. With these options, I can't select any region of the photo.


https://github.com/bluesky-social/social-app/assets/153161762/98520c09-3430-4be1-8118-8520fb2c9901

If I set the height and with to 250 though, I can select anywhere (which is what we want for avatars)


https://github.com/bluesky-social/social-app/assets/153161762/3c17d731-928b-4ff4-8713-e1962152af53